subsys/bluetooth/host/gatt.c

```diff
@@ -1205,7 +1205,9 @@ ssize_t bt_gatt_attr_write_ccc(struct bt_conn *conn,
 		return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
 	}
 
-	value = sys_get_le16(buf);
+	memset(&value, 0, sizeof(value));
+	memcpy(&value + offset, buf, len);
```

Comment:
This needs to be:
```mempy((u8_t *)&value + offset, buf, len);```

Reply:
I guess this function could just treat any non-zero offset as invalid and return BT_ATT_ERR_INVALID_OFFSET for that

Reply:
Same for `len == 0' actually (return BT_ATT_ERR_INVALID_ATTRIBUTE_LEN)

Reply:
Yep, that sounds better indeed. It beats me why someone would send just one byte for a CCC though.

Reply:
So something like:
```
if (offset > 0) {
        return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
}

if (!len || len > sizeof(u16_t)) {
        return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
}

if (len < sizeof(u16_t)) {
        value = *(u8_t *)buf;
} else {
        value = sys_get_le16(buf)
}
``` 

Thoughts?

Reply:
Yes, sounds good.